analysis/semsim/unifiedmodel.py

```diff
@@ -82,3 +88,83 @@ def _tokenize(self, example):
                 features[f"{example_key}_{feat_key}"] = features.pop(feat_key)
             features_full.update(features)
         return features_full
+
+
+class RRUM(pl.LightningModule):
+    def __init__(self, with_transcript, optimizer_config, freeze_policy=None, pos_weight=None):
+        super().__init__()
+        self.optimizer_config = optimizer_config
```

Comment:
Maybe we could refine the optimizer configuration little bit? Here at init we could add
```
self.optimizer_class = getattr(torch.optim, optimizer)
self.lr = lr
```
and add `optimizer` and `lr` to init arguments. The value of `optimizer` could be for example "Adam"

Reply:
Ah forgot to think about experimenting with LR schedulers, good point! So if we wanted experiment with LR schedulers it could be easier when using your way of optimizer configuration. As an example that could work something like this I guess:
```
optimizer_config=(lambda x: {"optimizer": torch.optim.Adam(x.parameters(), lr=lr), "lr_scheduler": torch.optim.lr_scheduler.SOMESCHEDULER()})
```